### PR TITLE
fix: add translation fixes for v2.4

### PR DESCRIPTION
**Pull request by jenniferarnesen**
Part of fix for: https://jira.dhis2.org/browse/DHIS2-8638

All i18n.t statements need to be executed at render time for the translations to be correct.

With Spanish (and Spanish-ish) translation:

Data Dimension:

image

image

image

image

Period dimension:
image

image

Chart title:
image

### CODE REVIEW

File: src/api/dimensions.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const selectFromResponse = (response, entity, selectorFn) =>
     typeof selectorFn === 'function' ? selectorFn(response) : response[entity]
 
 // Request functions
-const request = (d2, entity, paramString, { selectorFn } = {}) => {
+const request = ({ d2, entity, paramString }, { selectorFn } = {}) => {
```
Comment (jenniferarnesen) [inline src/api/dimensions.js:12]: Changes to `request` and `requestWithPaging` were required by eslint. (max params)

File: src/components/DataDimension/DataDimension.js [hunk 2]
```diff
@@ -136,13 +136,15 @@ export class DataDimension extends Component {
...
         if (augmentFn) {
-            dimensionItems = augmentFn(dimensionItems, groupId)
+            augmentedDimensionItems = augmentFn(dimensionItems, groupId)
         }
 
         const items = concatItems
-            ? this.state.items.concat(dimensionItems)
-            : dimensionItems
+            ? this.state.items.concat(augmentedDimensionItems)
+            : augmentedDimensionItems
```
Comment (jenniferarnesen) [inline src/components/DataDimension/DataDimension.js:145]: eslint fix only (use const for nextPage)

File: src/components/DataDimension/Detail.js [hunk 3]
```diff
@@ -4,10 +4,16 @@ import InputLabel from '@material-ui/core/InputLabel'
 import MenuItem from '@material-ui/core/MenuItem'
 import Select from '@material-ui/core/Select'
 import i18n from '@dhis2/d2-i18n'
+import { TOTALS, DETAIL } from '../../modules/dataTypes'
 
 import { styles } from './styles/Details.style'
 
-export const Detail = ({ value, onDetailChange, detailAlternatives }) => (
+const getOptions = () => ({
```
Comment (jenniferarnesen) [inline src/components/DataDimension/Detail.js:11]: Moved this object from dataTypes to here. This was a case of over-engineering, as the Detail component is only used in this one case, where the options are Totals and details.

File: src/components/DimensionsPanel/List/DimensionItem.js [hunk 4]
```diff
@@ -46,8 +45,7 @@ export class DimensionItem extends Component {
                 draggable={!isDeactivated}
                 onDragStart={onDragStart}
             >
-                {/* is it needed here or displayName should be used instead?! */}
-                {i18n.t(name)}
+                {name}
```
Comment (jenniferarnesen) [inline src/components/DimensionsPanel/List/DimensionItem.js:48]: This is a value coming from the backend and shouldn't be used with i18n.

File: src/components/DataDimension/DataDimension.js [hunk 5]
```diff
@@ -136,13 +136,15 @@ export class DataDimension extends Component {
             })) || DEFAULT_ALTERNATIVES
 
         const augmentFn = dataTypes[dataType].augmentAlternatives
+        let augmentedDimensionItems = dimensionItems
```
Comment (martinkrulltott) [inline src/components/DataDimension/DataDimension.js]: Use `const augmentedDimensionItems = augmentFn ? augmentFn(dimensionItems, groupId) : dimensionItems` instead?

**Review by martinkrulltott — APPROVED** [review]

File: src/components/DataDimension/DataDimension.js (hunk 5 above)
Comment (jenniferarnesen) [reply src/components/DataDimension/DataDimension.js]: fixed

**Pull request merged**